htmltools/_core.py

```diff
@@ -570,6 +573,18 @@ def __copy__(self: TagT) -> TagT:
         cp.__dict__.update(new_dict)
         return cp
 
+    def __enter__(self) -> None:
+        self.displayhook_stack.append(sys.displayhook)
+        sys.displayhook = self._displayhook
```

Comment:
This isn't directly relevant today, but looking at this code made me realize something. If we ever were to have Shiny run multithreaded, then we'd need locks around operations like this.

Reply:
Good point. I think when we get there, then we replace the displayhook at the beginning of process start with a function that defers to a contextvar equivalent of displayhook, and then all the rest of our code manipulates that contextvar instead of displayhook itself.

Reply:
Or... we could do that now I suppose?

Reply:
Actually let's not do it now, this is a big enough project already